kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiArrowBasedResultSet.java

```diff
@@ -34,6 +34,7 @@
 import org.apache.kyuubi.jdbc.hive.arrow.ArrowUtils;
 
 /** Data independent base class which implements the common part of all Kyuubi result sets. */
+@SuppressWarnings("deprecation")
 public abstract class KyuubiArrowBasedResultSet implements SQLResultSet {
```

Comment:
can you point to me why it would trow `deprecation` warning ?

Reply:
As `KyuubiArrowBasedResultSet` implements the deprecated method `getBigDecimal(int columnIndex, int scale` of `ResultSet` interface,
```
    @Deprecated
    BigDecimal getBigDecimal(int columnIndex, int scale) throws SQLException;
```

And we are getting warning when compiling `KyuubiArrowBasedResultSet`
```
Warning:  /home/runner/work/kyuubi/kyuubi/kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiArrowBasedResultSet.java:[106,20] [deprecation] getBigDecimal(String,int) in ResultSet has been deprecated
```

Similar reasons for adding `@SuppressWarnings("deprecation")` on classes in this PR.